manual/query_builder/README.md

```diff
@@ -0,0 +1,180 @@
+```
+
+When your query is complete, you can either extract a raw query string, or turn it into a
+[simple statement](../core/statements/simple) (or its builder):
+
+```java
+String cql = select.asCql();
+SimpleStatement statement = select.asSimpleStatement();
+SimpleStatementBuilder builder = select.asSimpleStatementBuilder();
+```
+
+#### Immutability
+
+All types in the fluent API are immutable. This means that every step creates a new object:
+
+```java
+SelectFrom selectFrom = selectFrom("ks", "user");
+
+Select select1 = selectFrom.column("first_name"); // SELECT first_name FROM ks.user
+Select select2 = selectFrom.column("last_name"); // SELECT last_name FROM ks.user
+
+assert select1 != select2;
+```
+
+Immutability has great benefits:
+
+* **thread safety**: you can share built queries across threads, without any race condition or
+  badly published state.
+* **zero sharing**: when you build multiple queries from a shared "base" (as in the example above),
+  all the queries are totally independent, changes to one query will never "pollute" another.
+  
+On the downside, immutability means that the query builder creates lots of short-lived objects.
+Modern garbage collectors are good at handling that, but still we recommend that you **avoid using
+the query builder in your hot path**:
+
+* favor [built statements](../core/statements/prepared) for queries that are used often. You can
+  still use the query builder and prepare the result:
+  
+  ```java
+  // During application initialization:
+  Select selectUser = selectFrom("user").all().where(isColumn("id").eq(bindMarker())); // SELECT * FROM ks.user WHERE id=?
+  PreparedStatement preparedSelectUser = session.prepare(selectUser.asSimpleStatement());
+
+  // At runtime:
+  session.execute(preparedSelectUser.bind(userId));
+  ```
+* for queries that never change, build them when your application initializes, and store them in a
+  field or constant for later.
+* for queries that are built dynamically, consider using a cache.
+
+#### Identifiers
+
+All fluent API methods use [CqlIdentifier] for schema element names (keyspaces, tables, columns...).
+But, for convenience, there are also `String` overloads if you use case-insensitive identifiers (as
+we recommend, see [Case sensitivity](../case_sensitivity) for more explanations).
+
+For conciseness, we'll use the string-based versions for the examples in this manual.
+
+### Non-goals
+
+The query builder is **NOT**:
+
+#### A crutch to learn CQL
+
+While the fluent API guides you, it does not encode every rule of the CQL grammar. Also, it supports
+a wide range of Cassandra versions, some of which may be more recent than your production target, or
+not even released yet. It's still possible to generate invalid CQL syntax if you don't know what
+you're doing.
+
+You should always start with a clear idea of the CQL query, and write the builder code that produces
+it, not the other way around.
+
+#### A better way to write static queries
```

Comment:
This is more opinionated, basically why I don't want to sacrifice type safety for prettier syntax.